Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -480,8 +481,12 @@ public extension PluginTarget {
                     // TODO: How do we determine what the executable name will be for the host platform?
                     return [.builtTool(name: target.name, path: RelativePath(target.name))]
                 }
+                else {
+                    return []
+                }
+            case .product(let product, _):
+                return [.builtTool(name: product.name, path: RelativePath(product.name))]
```

Comment:
This is the actual fix.  For a product we need the relative path of the produced executable, which will be the same as the product name.  The plugin itself will add any platform specific suffix, such as `.exe` on Windows.